### ISSUE TITLE
window icon on macos

### ISSUE THREAD

**Issue by samsabed** [iced-rs/iced#1467]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Is this issue related to iced?

- [X] My hardware is compatible and my graphics drivers are up-to-date.

### What happened?

using git head

```
let icon = Icon::from_file_data(icon_data, Some(image::ImageFormat::Png));
```

returns

```
Ok(
    Icon(
        NoIcon,
    ),
)
```

from `https://github.com/rust-windowing/winit` 

```
/// For platforms which don't have window icons (e.g. web)
#[derive(Debug, Clone, PartialEq, Eq)]
pub(crate) struct NoIcon;

    impl NoIcon {
        pub fn from_rgba(rgba: Vec<u8>, width: u32, height: u32) -> Result<Self, BadIcon> {
            // Create the rgba icon anyway to validate the input
            let _ = RgbaIcon::from_rgba(rgba, width, height)?;
            Ok(NoIcon)
        }
    }
```

rustc 1.64.0 (a55dd71d5 2022-09-19)

### What is the expected behavior?

```
let icon = Icon::from_file_data(icon_data, Some(image::ImageFormat::Png));
```

returns an icon and the window shows it

### Version

master

### Operative System

macOS

### Do you have any log output?

```shell
Ok(
    Icon(
        NoIcon,
    ),
)
```

**Comment by thunderstorm010** [comment]
TL;DR https://docs.rs/iced/latest/iced/window/struct.Settings.html#structfield.icon

You must pass that icon when you're starting your Application or Sandbox instance.
Example code:
```rs
// Assume App implements iced::Application
struct App;

fn main() {
   <App as iced::Application>::run(iced::Settings {
         window: iced::window::Settings {
                icon: Some(Icon::from_file_data(include_bytes!("../assets/logo.png"), Some(image::ImageFormat::Png)).expect("Failed to load icon")),
                ..Default::default()
        },
       ..Default::default()
    });
}
```
